lib/active_merchant/billing/gateways/hps.rb

```diff
@@ -296,21 +337,22 @@ def commit(action, &request)
         )
       end
 
+      SUCCESSFUL_RESPONSE_CODES = %w(0 00 85)
       def successful?(response)
         (
           (response['GatewayRspCode'] == '0') &&
-          ((response['RspCode'] || '00') == '00' || response['RspCode'] == '85')
+          ((SUCCESSFUL_RESPONSE_CODES.include? response['RspCode']) || !response['RspCode'])
         )
       end
 
       def message_from(response)
         if response['Fault']
           response['Fault']
         elsif response['GatewayRspCode'] == '0'
-          if response['RspCode'] != '00' && response['RspCode'] != '85'
-            issuer_message(response['RspCode'])
-          else
+          if SUCCESSFUL_RESPONSE_CODES.include? response['RspCode']
```

Comment:
Setting `SUCCESSFUL_RESPONSE_CODES` in one place makes this all a lil' more readable 👍